### ISSUE TITLE
Resolve not invoked with lazy-loaded module using loadChildren as promise

### ISSUE THREAD

**Issue by FedC** [angular/angular#37625]
# 🐞 bug report

### Affected Package
RouterModule from @angular/router

### Is this a regression?
I am running Angular 8 and I can reproduce in Angular 9

### Description
When the feature module is lazy loaded, the resolver never gets called.

```
const routes: Routes = [
  {
    path: 'second',
    loadChildren: () => import('/second/second.module')
      .then(m => m.SecondModule),
    resolve: {
      data: SecondResolverService
    }
  },
```

I tried pre-loading and eager-loading the feature module, but the resolver is never invoked.

I've searched the issues but I cannot find anything related to this issue I'm seeing. I did find a stack overflow answer that said it *should* be working, but I can reproduce the issue fairly easily. (see "Minimal Reproduction" below)

This is the stackoverflow answer: https://stackoverflow.com/questions/53346976/resolver-on-lazy-loading-angular

Not sure if this is intended to be this way, if it is, it would be nice to state in the documentation.
https://angular.io/api/router/Resolve
https://angular.io/guide/lazy-loading-ngmodules#verify-lazy-loading
https://angular.io/guide/router#lazy-loading-route-configuration
https://angular.io/guide/router#preloading-background-loading-of-feature-areas
From what I read, there is never an example where a resolver is used with lazy-loading module routes. But it's also never explicitly stated that we should NOT use it this way, either.

## 🔬 Minimal Reproduction
Here is the stackblitz where the issue is reproducible: 
https://stackblitz.com/edit/angular-ivy-ubes1q?file=src%2Fapp%2Fapp.module.ts

(The child module is lazy loaded, the resolver never gets called, which is proven by the child guard in child module routes.)

## 🔥 Exception or Error
none

## 🌍  Your Environment

**Angular Version:**
8 and 9

**Anything else relevant?**

I did see this issue that maybe is related: #36193 but as the tags say it's "confusing"

**Comment by FedC** [comment]
I did see this issue that maybe is related: https://github.com/angular/angular/issues/36193

**Comment by atscott** [comment]
This is actually working as intended. Since your guard returns false, the navigation is cancelled. The resolve stage happens after the guards stage. This is documented in the following places:
https://angular.io/api/router/Event

**Comment by FedC** [comment]
Thanks for the reply and your work @atscott !
I don't really understand why you say the guard would return false in this case, though since I didn't include a guard in the route, just a resolver.

**Comment by atscott** [comment]
> Thanks for the reply and your work @atscott !
> I don't really understand why you say the guard would return false in this case, though since I didn't include a guard in the route, just a resolver.

The child route that you're activating has a guard which returns false. Note that _all_ guards are performed before the resolvers. The execution is not guard for parent config -> resolver for parent config -> guard for child config -> resolver for child config but rather guard for parent config -> guard for child config -> resolver for parent config -> resolver for child config

**Comment by FedC** [comment]
I understand now. Thank you for clearing that up

**Comment by atscott** [comment]
Resolved by #38079, particularly this sentence in the [docs](https://angular.io/api/router/Resolve#usage-notes) for `Resolve`: 

> When both guard and resolvers are specified, the resolvers are not executed until all guards have run and succeeded. For example, consider the following route configuration:
